src/librustdoc/clean/types.rs

```diff
@@ -140,6 +140,22 @@ impl Item {
         name: Option<Symbol>,
         kind: ItemKind,
         cx: &DocContext<'_>,
+    ) -> Item {
+        Self::from_def_id_and_attrs_and_parts(
+            def_id,
+            name,
+            kind,
+            box cx.tcx.get_attrs(def_id).clean(cx),
```

Comment:
How common is the case where attrs are empty? That box can be omitted (by turning the field into Option<Box<>>). I was planning on doing that, but unless it's a bigger change, it could probably be part of this PR.

Reply:
I wouldn't bet on that to be honest. Trying to be too clever never ends well.

Reply:
We'll see, I guess I'll submit a PR anyway. But true, the additional complexity might backfire a bit.

Reply:
I don't think I'd call that too clever. @bugadani I'd be interested in seeing that PR :)